### PR TITLE
set env file and docker-compose.yml and made it working

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
**Description**

docker-compose.yml has been setup and works correctly with docker-compose up or npm start

User has to setup a personal mongodb instance and connect it with a mongo database by adding the mongo url in a .env file

this also ensures security in the development mode and best practices also

**Related Issue** 
- issue goes here

**Type of Change:**

- [x] Code
- [x] New Feature
- [x] Documentation

**How Has This Been Tested?**
 
 
image

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 pls check..
You just need to add a cluster  mongourl to your .env file and docker-compose up will also work

I've added volumes and network for linking in the docker file

File: src/app.module.ts [hunk 1]
```diff
@@ -15,7 +18,7 @@ import * as path from 'path';
     ),
     FirebaseModule,
     AuthModule,
-    MongooseModule.forRoot('mongodb://localhost/nest'),
+    MongooseModule.forRoot(process.env.MONGOURL),
```
Comment (Abhishek-kumar09) [inline src/app.module.ts:21]: create `.env.example` and put the environment variables with their example values.
like done here: https://github.com/codeforcauseorg/edu-client/blob/main/.env.example

Comment (Devesh21700Kumar) [reply src/app.module.ts:21]: ok..
sure

Comment (Devesh21700Kumar) [reply src/app.module.ts:21]: @Abhishek-kumar09 Done.. thanks for the review

**Comment by Abhishek-kumar09** [comment]
I have checked your PR, it is running perfectly, just finding some ways to run mongodb locally without cluster creation.

**Pull request closed without merging**